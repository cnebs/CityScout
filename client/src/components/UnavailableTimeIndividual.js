import React from 'react';
import { ListItem, ListItemText } from '@material-ui/core';
import moment from 'moment';
import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';

export default function UnavailableTimeIndividual({ time, handleDelete }) {
  const dayOfTheWeek = moment(time.time_start).format('dddd');
  const timeStart = moment(time.time_start).format('hh:mm a');
  const timeEnd = moment(time.time_end).format('hh:mm a');
  return (
    <ListItem>
      <ListItemText
        primary={time.name}
        secondary={`${dayOfTheWeek}s: ${timeStart} - ${timeEnd}`}
      ></ListItemText>
      <DeleteRoundedIcon onClick={() => handleDelete(time.item_id)} />
    </ListItem>
  );
}
