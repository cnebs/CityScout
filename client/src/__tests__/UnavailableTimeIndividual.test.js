import { shallow } from 'enzyme';
import React from 'react';
import UnavailableTimeIndividual from '../components/UnavailableTimeIndividual';
import { ListItem, ListItemText } from '@material-ui/core';
import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';

describe('UnavailableTimeIndividual component', () => {
  const handleDelete = jest.fn();
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(
      <UnavailableTimeIndividual
        time={{
          time_start: '2019-08-28T01:55:00.000Z',
          time_end: '2019-08-28T01:55:00.000Z',
          name: 'test',
          item_id: 1
        }}
        handleDelete={handleDelete}
      />
    );
  });
  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('it renders the corect child components', () => {
    expect(wrapper.find(ListItemText).length).toBe(1);
    expect(wrapper.find(ListItem).length).toBe(1);
    expect(wrapper.find(DeleteRoundedIcon).length).toBe(1);
  });
});
