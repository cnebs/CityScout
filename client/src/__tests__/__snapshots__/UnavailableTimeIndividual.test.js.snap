// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnavailableTimeIndividual component it renders without crashing 1`] = `ShallowWrapper {}`;
